guvnor-ala/guvnor-ala-build-maven/src/test/java/org/guvnor/ala/build/maven/executor/MavenCliOutputTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2016 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.ala.build.maven.executor;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.project.MavenProject;
+import org.guvnor.ala.build.Project;
+
+import org.guvnor.ala.build.maven.model.impl.MavenProjectImpl;
+import org.guvnor.ala.build.maven.util.MavenBuildExecutor;
+
+import org.guvnor.ala.build.maven.util.RepositoryVisitor;
+import org.guvnor.ala.source.Source;
+import org.guvnor.ala.source.git.GitHub;
+import org.guvnor.ala.source.git.GitRepository;
+import org.junit.After;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.scanner.embedder.MavenProjectLoader;
+
+/**
+ *
+ */
+public class MavenCliOutputTest {
+
+    public MavenCliOutputTest() {
+    }
+
+    private File tempPath;
+
+    @Before
+    public void setUp() throws IOException {
+        tempPath = Files.createTempDirectory("zzz").toFile();
+    }
+
+    @After
+    public void tearDown() {
+        FileUtils.deleteQuietly(tempPath);
+    }
+
+    @Test
+    public void waitForAppBuildTest() {
```

Comment:
Am I correct in assuming that this only tests the `MavenBuilderExecutor.executeMaven` method? If that is the case I think a unit test with a mocked Maven executor would be easier to understand and maintain.